lms/djangoapps/ccx/overrides.py

```diff
@@ -130,23 +137,33 @@ def override_field_for_ccx(ccx, block, name, value):
     field = block.fields[name]
     value_json = field.to_json(value)
     serialized_value = json.dumps(value_json)
-    try:
-        override = CcxFieldOverride.objects.create(
-            ccx=ccx,
-            location=block.location,
-            field=name,
-            value=serialized_value
-        )
-    except IntegrityError:
-        transaction.commit()
-        override = CcxFieldOverride.objects.get(
-            ccx=ccx,
-            location=block.location,
-            field=name
-        )
+    has_changes = False
+
+    override = get_override_for_ccx(ccx, block, name + "_instance")
+    if override:
+        has_changes = serialized_value != override.value
+
+    if not override:
+        try:
+            override = CcxFieldOverride.objects.create(
+                ccx=ccx,
+                location=block.location,
+                field=name,
+                value=serialized_value
+            )
+            _get_overrides_for_ccx(ccx).setdefault(block.location, {})[name + "_id"] = override.id
+        except IntegrityError:
+            transaction.commit()
```

Comment:
I think this was an effort to avoid a race condition with saving schedule changes (maybe because they were slow?). Unfortunately it was part of an omnibus commit, https://github.com/edx/edx-platform/commit/b3da2a54bb900f38c1049d91ddb5e4290beb1c8c 

@bdero, @cguardia do you remember more about this?